### ISSUE TITLE
Web components cannot be typechecked in templates with the defineCustomElement function

### ISSUE THREAD

**Issue by adrienZ** [vuejs/docs#2314]
the guide https://vuejs.org/guide/extras/web-components.html#building-custom-elements-with-vue may not be detailed enought for library authors.

creating a custom elements `defineCustomElement` will not be enought to allow [type checking on vue template](https://vuejs.org/guide/typescript/overview.html#typescript-in-templates)

I managed to do it by extending vue module types (as a believe this whats Nuxt does)
```
import Counter from './src/components/counter.ce.vue'

declare module 'vue' {
  export interface GlobalComponents {
    'web-counter': typeof Counter,
    'WebCounter': typeof Counter,
  }
}
``` 

[source code](https://github.com/adrienZ/example-vue-web-components-react/blob/main/packages/vue-ui/main.ts#L24)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/10186993/230639348-1a19ffef-20b5-4baf-8dda-44e1cab34b10.png)

Should we add this in the guide ?

**Comment by NataliaTepluhina** [comment]
@adrienZ yes, we can add this to custom elements guide. Would you be willing to create a pull request?

**Comment by adrienZ** [comment]
Hi @NataliaTepluhina, I can make one next week :)

**Comment by yyx990803** [comment]
closed via #2377
